### PR TITLE
Backport for #20136: Fixed bug where VolumeAmplification setting would not be applied correctly

### PR DESCRIPTION
**Pull request by biner73**
Backport to Matrix of issue described in #20136

### CODE REVIEW

**Comment by biner73** [comment]
@fuzzard

**Comment by howie-f** [comment]
sorry, i had to set the milestone to 19.3 cause 19.2 is just imminent

File: xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp [hunk 1]
```diff
@@ -484,12 +484,14 @@ bool CVideoPlayerAudio::ProcessDecoderOutput(DVDAudioFrame &audioframe)
       if (!m_audioSink.Create(audioframe, m_streaminfo.codec, m_synctype == SYNC_RESAMPLE))
         CLog::Log(LOGERROR, "%s - failed to create audio renderer", __FUNCTION__);
 
-      m_audioSink.SetDynamicRangeCompression((long)(m_processInfo.GetVideoSettings().m_VolumeAmplification * 100));
-
       if (m_syncState == IDVDStreamPlayer::SYNC_INSYNC)
         m_audioSink.Resume();
     }
-
+
```
Comment (howie-f) [inline xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp:490]: tab

File: xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp [hunk 2]
```diff
@@ -484,12 +484,14 @@ bool CVideoPlayerAudio::ProcessDecoderOutput(DVDAudioFrame &audioframe)
...
-
       if (m_syncState == IDVDStreamPlayer::SYNC_INSYNC)
         m_audioSink.Resume();
     }
-
+	
+    // Apply VolumeAmplification from settings on playback
+    m_audioSink.SetDynamicRangeCompression(
+	    static_cast<long>(m_processInfo.GetVideoSettings().m_VolumeAmplification * 100));
+
```
Comment (howie-f) [inline xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp:494]: tabs on both lines

**Review by howie-f — COMMENTED** [review]
please remove the tabs from these lines and only use spaces for indentation. thanks

File: xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp (hunk 2 above)
Comment (biner73) [reply xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp]: I originally had spaces per the coding guidelines, so not sure why it added tabs, but should be fixed now.

Comment (howie-f) [reply xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp]: all good :) could be your IDE does it automatically. there still seems to be something funky in your latest commit, jenkins points out

Comment (biner73) [reply xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp]: I did see that, it looks like only some of the target platform builds were successful, I'm not sure why some failed, but looking at the error logs it doesn't look related to my changes. Is there a way to get Jenkins to rerun the build tasks?

Comment (howie-f) [reply xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp]: yeah, don‘t worry about the build failures. i‘ll close and reopen this pr now so the build should rerun. please avoid to push changes to your branch while an active build is running. also, after the style changes are applied, the pr should be squashed down to a single commit… but first things first

Comment (biner73) [reply xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp]: Got it, I've squashed the commits down into one, we should be all good now. :)

**Comment by howie-f** [comment]
almost there. you can amend the commit to fix this up

**Comment by howie-f** [comment]
looks good now. one hint for the commit message: it should give a short explanation what (technically) changed with this commit. so you can get a better idea when browsing history. it's not always simple to find something good, for this particular case i'd recommend maybe: `set dynamic range compression for all audio frames`.. (the _Changed tabs to spaces_ leftover from the squash can be dropped) ... if you figure something even better please go ahead.

**Review by howie-f — APPROVED** [review]
thanks much for your contribution again 👍

**Comment by phunkyfish** [comment]
I think this can be merged for 19.2. It’s been tested in master, proving the fix is correct.

Any objections to changing the milestone and merging this?

**Comment by howie-f** [comment]
well, it is not a risky fix here but i think as we just purged the 19.2 milestone we should merge these shortly after the release happened

**Comment by phunkyfish** [comment]
Well if the train has left the station…

**Pull request merged**